diff -r -U 5 phpBB2++/common.php phpBB2++++/common.php
--- phpBB2++/common.php
+++ phpBB2++++/common.php
@@ -180,10 +180,17 @@
 {
 	header('Location: ' . $phpbb_root_path . 'install/install.' . $phpEx);
 	exit;
 }
 
+include($phpbb_root_path . 'feeds/config.'.$phpEx);
+
+if( !defined("FEEDS_INSTALLED") )
+{
+	message_die(GENERAL_ERROR, 'Feeds Syndicator is not installed correctlly.', '', __LINE__, __FILE__);
+}
+
 include($phpbb_root_path . 'includes/constants.'.$phpEx);
 include($phpbb_root_path . 'includes/template.'.$phpEx);
 include($phpbb_root_path . 'includes/sessions.'.$phpEx);
 include($phpbb_root_path . 'includes/auth.'.$phpEx);
 include($phpbb_root_path . 'includes/functions.'.$phpEx);
Only in phpBB2++/includes: functions.php.orig
diff -r -U 5 phpBB2++/index.php phpBB2++++/index.php
--- phpBB2++/index.php
+++ phpBB2++++/index.php
@@ -262,10 +262,36 @@
 	//
 	$is_auth_ary = array();
 	$is_auth_ary = auth(AUTH_VIEW, AUTH_LIST_ALL, $userdata, $forum_data);
 
 	//
+	// RSS Autodiscovery
+	//
+	$nav_links['alternate'] = array(
+		'all' => array(
+			'url'   => $feeds_config['autodiscovery_interface'],
+			'type'  => 'application/rss+xml',
+			'title' => $board_config['sitename']
+		)
+	);
+	if ( $viewcat != -1 )
+	{
+		for($i = 0; $i < $total_categories; $i++)
+		{
+			if ( $viewcat == $category_rows[$i]['cat_id'] )
+			{
+				$nav_links['alternate']['category'] = array(
+					'url'   => $feeds_config['autodiscovery_interface'] . "?" . POST_CAT_URL . "=$viewcat",
+					'type'  => 'application/rss+xml',
+					'title' => $category_rows[$i]['cat_title']
+				);
+			}
+		}
+		
+	}
+
+	//
 	// Start output of page
 	//
 	define('SHOW_ONLINE', true);
 	$page_title = $lang['Index'];
 	include($phpbb_root_path . 'includes/page_header.'.$phpEx);
Only in phpBB2++/install/schemas: mysql_basic.sql.orig
Only in phpBB2++: privmsg.php.orig
Only in phpBB2++/templates/subSilver: posting_body.tpl.orig
diff -r -U 5 phpBB2++/viewforum.php phpBB2++++/viewforum.php
--- phpBB2++/viewforum.php
+++ phpBB2++++/viewforum.php
@@ -382,10 +382,29 @@
 	'url' => append_sid('index.'.$phpEx),
 	'title' => sprintf($lang['Forum_Index'], $board_config['sitename'])
 );
 
 //
+// RSS Autodiscovery
+//
+$nav_links['alternate'] = array(
+	'all' => array(
+		'url'   => $feeds_config['autodiscovery_interface'],
+		'type'  => 'application/rss+xml',
+		'title' => $board_config['sitename']
+	)
+);
+if ( !in_array($forum_id, $feeds_config['exclude_forums']) )
+{
+	$nav_links['alternate']['forum'] = array(
+		'url'   => $feeds_config['autodiscovery_interface'] . "?" . POST_FORUM_URL . "=$forum_id",
+		'type'  => 'application/rss+xml',
+		'title' => $forum_row['forum_name']
+	);
+}
+
+//
 // Dump out the page header and load viewforum template
 //
 define('SHOW_ONLINE', true);
 $page_title = $lang['View_forum'] . ' - ' . $forum_row['forum_name'];
 include($phpbb_root_path . 'includes/page_header.'.$phpEx);
diff -r -U 5 phpBB2++/viewtopic.php phpBB2++++/viewtopic.php
--- phpBB2++/viewtopic.php
+++ phpBB2++++/viewtopic.php
@@ -533,10 +533,29 @@
 $nav_links['up'] = array(
 	'url' => $view_forum_url,
 	'title' => $forum_name
 );
 
+//
+// RSS Autodiscovery
+//
+$nav_links['alternate'] = array(
+	'all' => array(
+		'url'   => $feeds_config['autodiscovery_interface'],
+		'type'  => 'application/rss+xml',
+		'title' => $board_config['sitename']
+	)
+);
+if ( !in_array($forum_id, $feeds_config['exclude_forums']) )
+{
+	$nav_links['alternate']['topic'] = array(
+		'url'   => $feeds_config['autodiscovery_interface'] . "?" . POST_TOPIC_URL . "=$topic_id",
+		'type'  => 'application/rss+xml',
+		'title' => $topic_title
+	);
+}
+
 $reply_img = ( $forum_topic_data['forum_status'] == FORUM_LOCKED || $forum_topic_data['topic_status'] == TOPIC_LOCKED ) ? $images['reply_locked'] : $images['reply_new'];
 $reply_alt = ( $forum_topic_data['forum_status'] == FORUM_LOCKED || $forum_topic_data['topic_status'] == TOPIC_LOCKED ) ? $lang['Topic_locked'] : $lang['Reply_to_topic'];
 $post_img = ( $forum_topic_data['forum_status'] == FORUM_LOCKED ) ? $images['post_locked'] : $images['post_new'];
 $post_alt = ( $forum_topic_data['forum_status'] == FORUM_LOCKED ) ? $lang['Forum_locked'] : $lang['Post_new_topic'];
 
